tests/models/test_measurements.py

```diff
@@ -170,6 +170,52 @@ def test_measurements_cannot_be_saved_twice(ptf_datastore):
                 session.delete(m2)
                 session.commit()
 
+def test_threshold_flagging(ptf_datastore):
+    # get a copy of a single measurement
+    # create three new measurements based off it, and
+    #   set one to pass, one to fail mark, one to fail delete
+    # set the ds measurements to be just these three, then save and commit
+    # check we had the desired results
+    # also check that we can remove the dict, not change prov, and get expected behavior
+
+    measurements = ptf_datastore.measurements
+    m = measurements[0]  # grab the first one as an example
+
+    m.provenance.parameters['thresholds']['negatives'] = 0.3
+    m.provenance.parameters['deletion_thresholds']['negatives'] = 0.5
+
+    m.disqualifier_scores['negatives'] = 0.1 # set a value that will pass both
+    assert m.compare_to_thresholds() == "ok"
+
+    m.disqualifier_scores['negatives'] = 0.4 # set a value that will fail one
+    assert m.compare_to_thresholds() == "bad"
+
+    m.disqualifier_scores['negatives'] = 0.6 # set a value that will fail both
+    assert m.compare_to_thresholds() == "delete"
+
+    # test what happens if we set deletion_thresholds to None
+    #   This should set deletion_threshold = threshold
+    m.provenance.parameters['deletion_thresholds'] = None
+
+    m.disqualifier_scores['negatives'] = 0.1 # set a value that will pass both
+    assert m.compare_to_thresholds() == "ok"
+
+    m.disqualifier_scores['negatives'] = 0.4 # set a value that will fail 
+    assert m.compare_to_thresholds() == "delete"
+
+    # test what happens if we set deletion_thresholds to an empty dict
+    #   This should cause it to be impossible to be marked "delete"
+    m.provenance.parameters['deletion_thresholds'] = {}
+
+    m.disqualifier_scores['negatives'] = 0.1 # set a value that will pass both
+    assert m.compare_to_thresholds() == "ok"
+
+    m.disqualifier_scores['negatives'] = 0.9 # set a value that will fail
+    assert m.compare_to_thresholds() == "bad"
+
+    # I'd also like to test that deletion_thresholds is a proper non-critical param
+    # so that it can be changed without affecting the provenance
+    # However, it seems like prov.id changes even when varying non-critical params
```

Comment:
I went with a version of option 1 (pending discussion/review):

I decided to put a method on `Measurer` `compare_measurement_to_thresholds(self, m)` which uses the non-critical deletion threshold dictionary of that measurer object and the `mark_threshold` dictionary from the Measurements object's provenance.

I wasn't sure the best spot to put it when I first made it a method of `Measurer`, but it could also have been a method of `Measurements` (in which case the deletion dictionary would need to be passed to it instead) or even a utility function (I didn't like this because it felt separated from the relevant code).

I will continue to ponder the best implementation.